Jetpack Manage: Display the selected licenses in the review licenses modal

yashwin opened the pull request:
Resolves https://github.com/Automattic/jetpack-genesis/issues/97

## Proposed Changes

This PR displays the selected licenses in the review licenses modal.

## Testing Instructions

**Prerequisites**

Since these changes are made specifically for agencies, you must set yourself(partner) as an agency - 2c49b-pb. Make sure to switch it back to the previous type.

**Instructions**

1. Visit the Jetpack Cloud link > Replace `dashboard` in the URL with `/partner-portal/issue-license?flags=jetpack/bundle-licensing`.
2. Select a few licenses and click the `Review X licenses` button > Verify that the selected licenses are displayed as shown below. Verify it displays well on all screen sizes.

Large screen view:

<img width="1123" alt="Screenshot 2023-11-23 at 9 22 37 AM" src="https://github.com/Automattic/wp-calypso/assets/10586875/52cb3276-2c86-4cbb-aa9a-fced9a809e5e">

Mobile view:

![mobile (43)](https://github.com/Automattic/wp-calypso/assets/10586875/98627c4a-3e4f-46fb-a630-6a48cd7f013b)

Tablet view:

![mobile (44)](https://github.com/Automattic/wp-calypso/assets/10586875/f42af07c-f0ae-40fa-b366-32413664c56b)


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

yashwin committed:
Display selected licenses in the license review screen

jkguidaven commented on client/jetpack-cloud/sections/partner-portal/issue-license-v2/review-licenses/index.tsx:
```diff
@@ -10,13 +17,25 @@ interface Props {
+						{ translate( 'You’re about to issue the following licenses:' ) }
+					</div>
+					<div className="review-licenses__selected-licenses">
+						{ selectedLicenses.map( ( license ) => (
```
It would be helpful if we could sort the list by product type. This way, it would be easier to view products of the same type in a group. Additionally, I am wondering if it would be possible to combine the product count so that each product type only occupies one space. 🤔 wdyt @jeffgolenski?

<img width="603" alt="Screen Shot 2023-11-23 at 4 50 07 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/3afa878b-b847-4f43-81a6-a09899a36eb0">

yashwin replied:
Good suggestion, @jkguidaven! I have updated the PR to group licenses by type & sort them by quantity.

> Additionally, I am wondering if it would be possible to combine the product count so that each product type only occupies one space. 🤔 wdyt @jeffgolenski?

I'm not sure how complicated it will be to decide how to combine the bundles. I think we can discuss this & address this in a different PR if needed. WDYT?

jkguidaven replied:
> I'm not sure how complicated it will be to decide how to combine the bundles. I think we can discuss this & address this in a different PR if needed. WDYT?

sounds good to me 👍

jeffgolenski replied:
@jkguidaven @yashwin Apologies, I was AFK for a holiday. I realize this has been merged and we'll pick up the convo in a different PR, but when you say: "view the products of the same type in the same group" and "and product type only occupies on space" I'm unsure what you mean. Can you elaborate?

jkguidaven replied:
Hi @jeffgolenski,

I apologize for not being clear earlier. In the screenshot example, we can see that there are two licenses for Akismet Anti-spam - one with 1 license and the other with 10 licenses.  I was thinking To simplify this, we can combine them into one line item and show Akismet Anti-spam with a total of 11 licenses.

<img width="603" alt="Screen Shot 2023-11-23 at 4 50 07 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/3afa878b-b847-4f43-81a6-a09899a36eb0">

I guess one potential concern is that it could be challenging for users to determine which bundles they have selected and the corresponding discounts they are receiving. But let me know what you think.

jeffgolenski replied:
@jkguidaven Ah good find my friend. For now, let's **keep them separate** since this is an itemized list. Hopefully users remember that they selected a bundle and a single, but I wouldn't rely on them if they're purchasing a lot. In any case, I suspect it may be an edge case to see this, but It's still too soon to know!

jkguidaven commented on client/jetpack-cloud/sections/partner-portal/issue-license-v2/review-licenses/license-info.tsx:
```diff
@@ -0,0 +1,41 @@
+
+	return (
+		<div className="review-licenses__selected-license">
+			<div className="review-licenses__details">
```
[non-blocker] I suggest that you add the prefix `review-licenses__selected-license` to the child elements such as `review-licenses__license-details` and `review-licenses__license-info`, provided that these child elements are not used elsewhere. This will help to clarify the relationship between the container and its child elements and make it easier for readers to understand if these are standalone elements.

jkguidaven reviewed (commented):
Nice work! I tested and can confirm that the functionality works as expected. 👍 

I have left one suggestion for feedback from design and one non-blocking comment.

yashwin committed:
Group licenses by type and rename className

yashwin commented:
Thanks for the review, @jkguidaven! I have addressed all your comments & made the following changes: https://github.com/Automattic/wp-calypso/pull/84452/commits/82f81dfb23740256b501e19762f048dde54e05fd

jkguidaven reviewed (approved):
Thanks for addressing my comments. LGTM! nice work 👍

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9736230

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @yashwin for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9736230).
